pkg/actions/addon/addon_test.go

```diff
@@ -1,19 +1,51 @@
 package addon_test
 
 import (
+	"context"
+
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
 	"github.com/weaveworks/eksctl/pkg/actions/addon"
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
-	"github.com/weaveworks/eksctl/pkg/eks/mocksv2"
+	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
 )
 
 var _ = Describe("Addon", func() {
-	When("the version is supported", func() {
-		It("does not error", func() {
-			_, err := addon.New(&api.ClusterConfig{Metadata: &api.ClusterMeta{Version: "1.18"}}, &mocksv2.EKS{}, nil, false, nil, nil)
-			Expect(err).NotTo(HaveOccurred())
+	var (
+		clusterConfig *api.ClusterConfig
+		mockProvider  *mockprovider.MockProvider
+		manager       *addon.Manager
+	)
+
+	BeforeEach(func() {
+		var err error
+		clusterConfig = &api.ClusterConfig{Metadata: &api.ClusterMeta{
+			Version: "1.18",
+			Name:    "my-cluster",
+		}}
+		mockProvider = mockprovider.NewMockProvider()
+		manager, err = addon.New(clusterConfig, mockProvider.EKS(), nil, false, nil, nil)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	When("addon is coredns with no nodegroup", func() {
+		It("does not wait for addon to be active", func() {
+			a := &api.Addon{
+				Name: api.CoreDNSAddon,
+			}
+			err := addon.ExportWaitForAddonToBeActive(manager, context.Background(), a, 0)
+			Expect(err).To(BeNil())
+		})
+	})
+
+	When("addon is aws-ebs-csi-driver with no nodegroup", func() {
+		It("does not wait for addon to be active", func() {
+			a := &api.Addon{
+				Name: api.AWSEBSCSIDriverAddon,
+			}
+			err := addon.ExportWaitForAddonToBeActive(manager, context.Background(), a, 0)
+			Expect(err).To(BeNil())
```

Comment:
This should go in `pkg/actions/addon/create_test.go` and, as noted in the comment above, it should be tested by calling `manager.Create`.